File: src/legacy/Bakabase.InsideWorld.Business/Components/PostParser/Models/Domain/PostParserTask.cs
using System.Collections.Generic;
using System.Text.Json.Nodes;
using Bakabase.InsideWorld.Business.Components.PostParser.Models.Domain.Constants;
using Bakabase.Modules.Workflow.Abstractions.Models.Domain.Constants;

namespace Bakabase.InsideWorld.Business.Components.PostParser.Models.Domain;

public record PostParserTask
{
    public int Id { get; set; }
    public PostParserSource Source { get; set; }
    public string Link { get; set; } = null!;
    public string? Title { get; set; }
    public string? Content { get; set; }
    public string? Text { get; set; }
    public int Revision { get; set; }
    public int? WorkflowDefinitionId { get; set; }
    public int? WorkflowRunId { get; set; }
    public WorkflowRunStatus? WorkflowStatus { get; set; }
    public List<PostParseTarget> Targets { get; set; } = [];
    [Newtonsoft.Json.JsonProperty(ItemConverterType = typeof(PostParserResultNodeConverter))]
    public Dictionary<PostParseTarget, JsonNode?>? Results { get; set; }
    public string? Error { get; set; }
    public bool IsDeleted { get; set; }
}

File: src/web/src/core/models/PostParserTask.ts
import type { PostParseTarget, PostParserSource, WorkflowRunStatus } from "@/sdk/constants";

export interface PostParserTask {
  id: number;
  source: PostParserSource | 0;
  link: string;
  text?: string | null;
  title?: string;
  content?: string;
  targets: PostParseTarget[];
  results?: Record<string | number, unknown>;
  error?: string;
  isDeleted?: boolean;
  workflowRunId?: number | null;
  workflowDefinitionId?: number | null;
  workflowStatus?: WorkflowRunStatus | null;
  revision?: number;
}
